libs/utils/android/screen.py

```diff
@@ -136,4 +137,25 @@ def get_screen_density(target):
         """
         return target.execute('getprop ro.sf.lcd_density')
 
+    @staticmethod
+    def is_screen_on(target):
+	cmdout = target.execute('dumpsys input_method')
+	return ('mInteractive=true' in cmdout)
+
+    @staticmethod
+    def set_screen_on(target):
+        log = logging.getLogger('Screen')
+        if Screen.is_screen_on(target) == False:
+            log.info('Screen is off, turning on')
+            System.power(target)
+
+    @staticmethod
+    def unlock(target):
+       Screen.set_screen_on(target)
+       target.execute('input keyevent 82')
+       # Incase screen was already unlocked, code 82
+       # may cause it zoom out, in this case just issue
+       # back key to restore it
+       System.back(target)
```

Comment:
Perhaps this is not working in case the scree is already on and an app runnning, we are not granted to be in the home screen.

I think this sequence should be more robust:
```python
System.set_screen(on=True)
System.menu()
System.home()
```

This should grant to unlock the device and move back to the home screen... at least it did for all the starting conditions I've tested.